lib/honeycomb/propagation/honeycomb.rb

```diff
@@ -58,8 +59,14 @@ def parse_v1(payload)
 
     # Serialize trace headers
     module MarshalTraceContext
+      # for cleaning data in trace fields before serializing to prop header value
+      include Libhoney::Cleaner
+      # promote cleaner instance methods to module methods so that self.to_trace_header can use them
+      module_function :clean_data, :clean_string
```

Comment:
As I was monkey-patching our code locally, I was lamenting that `Libhoney::Cleaner` didn't already make these `module_function`s, so that I could just call `Libhoney::Cleaner.clean_data` as a singleton. 😛 But yeah, this looks about how I expected it to look. (Our monkey-patch goes through `add_trace_field` instead, cuz it was marginally easier to hack around, but I do think the cleaning properly belongs in the propagation code. 👍)

Reply:
Yeah, totally. I'd do the same. Fewer gems to bump, lol.